cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -153,6 +154,7 @@
                           RunRecordMonitorService runRecordMonitorService) {
     this.maxConcurrentRuns = cConf.getInt(Constants.AppFabric.MAX_CONCURRENT_RUNS);
     this.maxConcurrentLaunching = cConf.getInt(Constants.AppFabric.MAX_CONCURRENT_LAUNCHING);
+    this.userProgramLaunchDisabled = cConf.getBoolean(Constants.AppFabric.USER_PROGRAM_LAUNCH_DISABLED, false);
```

Comment:
better to also add this to cdap-default.xml with false

Reply:
yeah I think it's still better. I think it's a valid use case to have an admin that wants to configure CDAP to only run user workloads on remote clusters and save the CDAP cluster for system stuff.